somafm.el

```diff
@@ -150,10 +181,25 @@ Refresh channels list if necessary."
   (somafm--show-channels-buffer)
   (move-beginning-of-line nil))
 
+(defun somafm--favorite-p (song)
+  "Check if SONG is in the favorites file."
+  (if (file-exists-p somafm-favorites-file)
+      (with-temp-buffer
+        (insert-file-contents somafm-favorites-file)
+        (goto-char (point-min))
+        (search-forward-regexp
+         (somafm--make-favorite-regexp song)
+         nil t))))
```

Comment:
We can use `when` here since there's no else branch
```suggestion
  (when (file-exists-p somafm-favorites-file)
    (with-temp-buffer
      (insert-file-contents somafm-favorites-file)
      (goto-char (point-min))
      (search-forward-regexp
       (somafm--make-favorite-regexp song)
       nil t)))
```